consensus/src/pos/input_data.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use chainstate_types::pos_randomness::PoSRandomness;
+use common::{
+    chain::block::timestamp::BlockTimestamp,
+    chain::{config::EpochIndex, PoolId, TxInput, TxOutput},
+    primitives::Amount,
+};
+use crypto::{
+    key::{PrivateKey, PublicKey},
+    vrf::{VRFPrivateKey, VRFPublicKey},
+};
+use serialization::{Decode, Encode};
+
+pub use crate::{
+    error::ConsensusVerificationError,
+    pos::{
+        block_sig::BlockSignatureError, check_pos_hash, error::ConsensusPoSError,
+        kernel::get_kernel_output, stake, target::calculate_target_required,
+        target::calculate_target_required_from_block_index, StakeResult,
+    },
+    pow::{calculate_work_required, check_proof_of_work, mine, ConsensusPoWError, MiningResult},
+    validator::validate_consensus,
+};
+
+/// Input needed (i.e provided by a wallet) to generate PoS consensus data
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub struct PoSGenerateBlockInputData {
+    /// The private key of the staker
+    stake_private_key: PrivateKey,
+    /// The VRF private key (i.e used for sealed epoch data)
+    vrf_private_key: VRFPrivateKey,
+    /// The pool Id of stake pool
+    pool_id: PoolId,
+    /// The transaction input for the block reward
+    kernel_inputs: Vec<TxInput>,
+    /// The transaction input UTXOs
+    kernel_input_utxos: Vec<TxOutput>,
+}
+
+impl PoSGenerateBlockInputData {
+    pub fn kernel_inputs(&self) -> &Vec<TxInput> {
+        &self.kernel_inputs
+    }
+
+    pub fn kernel_input_utxos(&self) -> &Vec<TxOutput> {
+        &self.kernel_input_utxos
+    }
+
+    pub fn pool_id(&self) -> PoolId {
+        self.pool_id
+    }
+
+    pub fn stake_private_key(&self) -> &PrivateKey {
+        &self.stake_private_key
+    }
+
+    pub fn stake_public_key(&self) -> PublicKey {
+        PublicKey::from_private_key(&self.stake_private_key)
+    }
+
+    pub fn vrf_private_key(&self) -> &VRFPrivateKey {
+        &self.vrf_private_key
+    }
+}
+
+#[derive(Debug, Clone, Encode, Decode)]
+/// Input needed to finalize PoS consensus data
```

Comment:
Please put the comments above derive macros.

Also, please explain with more information. Who provides this information and in what circumstance (same for the other struct).

Reply:
I've expanded here, which should hopefully be what wallet RPC devs to know what to pass in. Let me know if any section needs further detail